src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -257,16 +265,49 @@ public function denormalize($data, $class, $format = null, array $context = arra
                 continue;
             }
 
-            $value = $this->validateAndDenormalize($class, $attribute, $value, $format, $context);
+            try {
+                $value = $this->validateAndDenormalize($class, $attribute, $value, $format, $context, $accumulatingContext);
+            } catch (NotDenormalizableValueException $exception) {
+                if ($shouldCollectAllErrors) {
+                    $accumulatingContext[$attribute][] = $exception;
+
+                    continue;
+                }
+
+                throw $exception;
+            } catch (UnexpectedValuesException $exception) {
+                $accumulatingContext[$attribute] = $exception->getUnexpectedValueErrors();
+
+                continue;
+            }
+
             try {
                 $this->setAttributeValue($object, $attribute, $value, $format, $context);
             } catch (InvalidArgumentException $e) {
-                throw new NotNormalizableValueException($e->getMessage(), $e->getCode(), $e);
+                $exception = new NotDenormalizableValueException($e->getMessage(), $attribute, $e);
+
+                if ($shouldCollectAllErrors) {
+                    $accumulatingContext[$attribute][] = $exception;
+
+                    continue;
+                }
+
+                throw $exception;
             }
         }
 
         if (!empty($extraAttributes)) {
-            throw new ExtraAttributesException($extraAttributes);
+            if (!$shouldCollectAllErrors) {
+                throw new ExtraAttributesException($extraAttributes);
```

Comment:
Throwing an exception looks not really good to me because it forces everybody to catch the exception and re-throw on any normalizer that decorate the AbstractObjectNormalizer.

Here is an example of normalizer we could imagine:

```php
class PhoneNumberNormalizer implements DenormalizerInterface
{
    public function __construct(DenormalizerInterface $objectNormalizerDecorated)
    {
        $this->decorated = $objectNormalizerDecorated;
    }

    public function denormalize($data, $class, $format = null, array $context = array(), AccumulatingContext $accumulatingContext = null)
    {
        $object = $this->decorated->denormalize($data);
        // Here the exception may be throw while we accumulate errors
        // We should catch and add a new throw (and for each normalizer)
        
        // I do some stuff with $object
        // Well, maybe we should re-throw here in case something's wrong?

        // Ok, I will re-throw (exceptions should be nested, which is another problem)
        // How am I suppose at this point know what attribute I want to add?

        // nevermind, the library is not up to date to support this feature and will generate weird behavior
    }
}
```

The exception is (IMHO) not designed for this kind of usage. I suggest the usage of a new concept: the context of serialization needs to be an object. And it should also contain an error object (something like your accumulatingContext). This way the user will be able to configure properly (and without only using constants) the serialization/deserialization and also will be able to get back information about what happened in the process.

This may be a lot of work but probably the real best solution. (what do you think @dunglas ?)

_Note that another solution I see would be to give the AccumulationContext as user instead of just a boolean. I posted this suggestion as example here: https://github.com/symfony/symfony/issues/28358_